tybalt/data_models.py

```diff
@@ -261,17 +270,25 @@ def nn(self, n_components, model='tybalt', transform_df=False, **kwargs):
                                    learning_rate=learning_rate,
                                    loss=loss,
                                    verbose=verbose,
+                                   tied_weights=tied_weights,
                                    optimizer=adage_optimizer)
             self.adage_fit.initialize_model()
             self.adage_fit.train_adage(train_df=self.nn_train_df,
                                        test_df=self.nn_test_df,
                                        adage_comparable_loss=adage_comp_loss)
-            self.adage_decoder_w = self.adage_fit.get_decoder_weights()
+            self.adage_decoder_w = (
+                self.adage_fit.get_weights(decoder=use_decoder_weights)
+                )
 
             features = ['dae_{}'.format(x) for x in range(0, latent_dim)]
-            self.adage_weights = pd.DataFrame(self.adage_decoder_w[1][0],
```

Comment:
Okay, checking my understanding here -- if `tied_weights` is true and it's an ADAGE model  ([line 157](https://github.com/greenelab/tybalt/pull/123/files#diff-96399e6706013365ab70b1d790e938fdR157)), `use_decoder_weights = False` and if that's the case, you use the same bit of code as you used to here in [line 285](https://github.com/greenelab/tybalt/pull/123/files#diff-96399e6706013365ab70b1d790e938fdR285). This is different from what I would expect/not intuitive for me given your initial comment in the PR, so wanted to just check at this point.

Reply:
Yeah, I agree that this is a bit confusing. Essentially, if `tied_weights=True` there, effectively, _are no_ decoder weights in an ADAGE model. Instead, they are transposed encoder weights.

I think a likely source of confusion is the variable naming here. I update this in 6b4dc0f